Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-239.ca-central-1.compute.internal
Sha: e6765ea24ade5a8be91922dac7d4cf2bc81e0f52
Number of Error(s): 3
make[2]: *** [esp/src/CMakeFiles/eclwatch.dir/build.make:308: esp/src/build/dist/stub.eclwatch.js] Error 1
make[1]: *** [CMakeFiles/Makefile2:10345: esp/src/CMakeFiles/eclwatch.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-01-07_14-29-53
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   366 sec (00:06:06)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   375 sec (00:06:15)
time logs end

[Build error:](https://gist.github.com/c4f4ca49a35fd2061ebb865b5c67bc0d)
Build log [Build log](https://gist.github.com/HPCCSmoketest/c4f4ca49a35fd2061ebb865b5c67bc0d#file-result-21-01-07-14-23-26-log)

--- a/esp/src/.eslintrc.js
+++ b/esp/src/.eslintrc.js
@@ -8,6 +8,7 @@ module.exports = {
     extends: [
         "eslint:recommended",
         "plugin:@typescript-eslint/recommended",
+        "plugin:react-hooks/recommended"
     ],
     env: {
         "browser": true,

--- a/esp/src/package.json
+++ b/esp/src/package.json
@@ -4,7 +4,7 @@
   "description": "'ECL Watch' Web interface for HPCC Platform.",
   "scripts": {
     "clean": "rimraf ./build ./lib ./src/nlsHPCCType.ts",
-    "lint": "eslint eclwatch/**/*.js src/**/*.ts?",
+    "lint": "eslint eclwatch/**/*.js src/**/*.ts? src-react/**/*.ts?",
     "lint-fix": "eslint --fix eclwatch/**/*.js src/**/*.ts",
     "copy-res-es6-promise": "cpx \"./node_modules/es6-promise/dist/es6-promise.auto.min.js\" ./build/node_modules/es6-promise/dist/",
     "copy-res-eclwatch-img": "cpx \"./eclwatch/img/**/*.{png,jpg,gif}\" ./build/eclwatch/img/",
@@ -77,7 +77,8 @@
     "cpx": "^1.5.0",
     "css-loader": "^3.4.2",
     "dojo-webpack-plugin": "^2.8.13",
-    "eslint": "^7.14.0",
+    "eslint": "^7.17.0",
+    "eslint-plugin-react-hooks": "^4.2.0",
     "file-loader": "^5.1.0",
     "local-web-server": "^4.0.0",
     "minimist": ">=1.2.2",

--- a/esp/src/src-react/layouts/DojoAdapter.tsx
+++ b/esp/src/src-react/layouts/DojoAdapter.tsx
@@ -31,8 +31,9 @@ export const DojoAdapter: React.FunctionComponent<DojoProps> = ({
     React.useEffect(() => {
 
         const elem = document.createElement("div");
-        myRef.current.innerText = "";
-        myRef.current.appendChild(elem);
+        const divRef = myRef.current;
+        divRef.innerText = "";
+        divRef.appendChild(elem);
 
         let widget = undefined;
 
@@ -78,12 +79,12 @@ export const DojoAdapter: React.FunctionComponent<DojoProps> = ({
                 });
                 //  ---
 
-                const domNode = ReactDOM.findDOMNode(myRef.current) as Element;
+                const domNode = ReactDOM.findDOMNode(divRef) as Element;
                 domNode.innerHTML = "";
             }
             widget = null;  //  Avoid race condition  ---
         };
-    }, [widgetClassID]);
+    }, [onWidgetMount, params, uid, widgetClass, widgetClassID]);
 
     return <div ref={myRef} style={{ width: "100%", height: "100%" }}>{nlsHPCC.Loading} {widgetClassID}...</div>;
 };

--- a/esp/src/src/react/vizComponent.tsx
+++ b/esp/src/src/react/vizComponent.tsx
@@ -27,7 +27,7 @@ export const VisualizationComponent: React.FunctionComponent<VisualizationProps>
         return () => {
             w.target(null);
         };
-    }, []);
+    }, [divID, widget]);
 
     if (widgetInstance) {
         if (widgetProps.columns) {

--- a/esp/src/src/react/wuStatus.tsx
+++ b/esp/src/src/react/wuStatus.tsx
@@ -85,7 +85,7 @@ export const WUStatus: React.FunctionComponent<WUStatus> = ({
         return () => {
             wuWatchHandle.release();
         };
-    }, []);
+    }, [wuid]);
 
     return (
         <MuiThemeProvider theme={theme}>

--- a/esp/src/webpack.config.js
+++ b/esp/src/webpack.config.js
@@ -21,7 +21,7 @@ proxyItems.forEach(item => {
 });
 
 module.exports = function (env) {
-    const isDev = env && env === "development";
+    const isDev = env?.development || env === "development";
     const isProduction = !isDev;
 
     const entry = {

